src/careamics/config/vae_algorithm_model.py

```diff
@@ -0,0 +1,118 @@
+"""Algorithm configuration."""
+
+from __future__ import annotations
+
+from pprint import pformat
+from typing import Literal, Optional, Union
+
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from typing_extensions import Self
+
+from .architectures import CustomModel, LVAEModel
+from .likelihood_model import GaussianLikelihoodModel, NMLikelihoodModel
+from .nm_model import GaussianMixtureNmModel
+from .optimizer_models import LrSchedulerModel, OptimizerModel
+
+
+class VAEAlgorithmConfig(BaseModel):
+    """Algorithm configuration.
+
+    This Pydantic model validates the parameters governing the components of the
+    training algorithm: which algorithm, loss function, model architecture, optimizer,
+    and learning rate scheduler to use.
+
+    Currently, we only support N2V, CARE, N2N and custom models. The `n2v` algorithm is
+    only compatible with `n2v` loss and `UNet` architecture. The `custom` algorithm
+    allows you to register your own architecture and select it using its name as
+    `name` in the custom pydantic model.
+
+    Attributes
+    ----------
+    algorithm : Literal["n2v", "custom"]
+        Algorithm to use.
+    loss : Literal["n2v", "mae", "mse"]
+        Loss function to use.
+    model : Union[UNetModel, LVAEModel, CustomModel]
+        Model architecture to use.
+    optimizer : OptimizerModel, optional
+        Optimizer to use.
+    lr_scheduler : LrSchedulerModel, optional
+        Learning rate scheduler to use.
+
+    Raises
+    ------
+    ValueError
+        Algorithm parameter type validation errors.
+    ValueError
+        If the algorithm, loss and model are not compatible.
+
+    Examples
+    --------
+    # TODO add once finalized
+    """
+
+    # Pydantic class configuration
+    model_config = ConfigDict(
+        protected_namespaces=(),  # allows to use model_* as a field name
+        validate_assignment=True,
+        extra="allow",
+    )
+
+    # Mandatory fields
+    # defined in SupportedAlgorithm
+    algorithm_type: Literal["vae"]
+    algorithm: Literal["musplit", "denoisplit", "custom"]
+    loss: Literal["musplit_loss", "denoisplit_loss"]
```

Comment:
In fact, maybe we should just type hint with the `Enum` classes instead of `Literal`. I've had some discussion with Joran about type hinting with `Literal` on user facing code and  using `Enum` internally, but maybe not using `Literal` will save us some trouble. 

Also, I just checked and the following code works, so users would be able to initialise the class with strings anyway.
```python
from pydantic import BaseModel, ConfigDict
from careamics.utils import BaseEnum

class Colour(str, BaseEnum):
    RED: str = "red"
    GREEN: str = "green"
    BLUE: str = "blue"

class Paint(BaseModel):
    # Pydantic class configuration
    model_config = ConfigDict(
        protected_namespaces=(),  # allows to use model_* as a field name
        validate_assignment=True,
        extra="allow",
    )

    colour: Colour

paint_instance = Paint(colour="green")
```

In this case though, there would have to be additional validation that the loss is only `musplit`, `denoisplit` or `custom` and not `mae` etc.